dbt/include/athena/macros/materializations/models/incremental/on_schema_change.sql

```diff
@@ -35,3 +42,50 @@
   {% endset %}
   {% do log(schema_change_message) %}
 {% endmacro %}
+
+{% macro athena__alter_column_type(relation, column_name, new_column_type) -%}
+  {#
+    1. Create a new column (w/ temp name and correct type)
+    2. Copy data over to it
+    3. Drop the existing column
+    4. Rename the new column to existing column
+  #}
+  {%- set tmp_column = column_name + "__dbt_alter" -%}
+
+  {# Fix bug in Athena: error when using quoted table name #}
+  {%- set relation = relation.render_pure() -%}
+
+  {# Fix bug in Athena: iceberg alter statements requires "string" column type instead of "varchar" #}
```

Comment:
does it make sense to use | improve existing macro `ddl_data_type` to avoid column types logic duplication ?

Reply:
I agree. I will fix it. I need additional time to test timestamp(6) replacement to timestamp to iceberg tables only. And I need to resolve situation when the last query fail. Next run will drop tmp column with data.